src/target/riscv/riscv-013.c

```diff
@@ -386,12 +386,12 @@ static void dump_field(int idle, const struct scan_field *field)
 	uint64_t out = buf_get_u64(field->out_value, 0, field->num_bits);
 	unsigned int out_op = get_field(out, DTM_DMI_OP);
 	unsigned int out_data = get_field(out, DTM_DMI_DATA);
-	unsigned int out_address = out >> DTM_DMI_ADDRESS_OFFSET;
+	unsigned int out_address = (out >> DTM_DMI_ADDRESS_OFFSET) - target->dbgbase;
```

Comment:
I'd recommend to dump the real address used on the DMI interface, not just the relative one to the base.

```suggestion
	unsigned int out_address = out >> DTM_DMI_ADDRESS_OFFSET;
```

Reply:
Oh, my mistake,  In the [previous version](https://github.com/riscv/riscv-openocd/commit/b2948d319e5768eb2543ec5c5ab1645c1c0ff496#diff-701864e2be3210275085c84b8f6fa7f9eb6171a2a18ca71638be42a155c08c63L403-L404), I modified the address after the address was dump and before the decode_dmi, as decode_dmi need relative address.
In the later version, when I delete `if (target->dbgbase_set)` , I forgot the address dump above and put the address change directly to the top. 
Thanks for pointing it out, I'll put it before decode_dmi, so we can dump the real address.